modin/engines/base/frame/partition_manager.py

```diff
@@ -245,12 +248,23 @@ def broadcast_axis_partitions(
         # may want to line to partitioning up with another BlockPartitions object. Since
         # we don't need to maintain the partitioning, this gives us the opportunity to
         # load-balance the data as well.
+        kw = {
+            "num_splits": num_splits,
+            "other_axis_partition": right_partitions,
+        }
+        if lengths:
+            kw.update(
+                {
+                    "_lengths": lengths,
+                    "manual_partition": True,
+                }
+            )
```

Comment:
```suggestion
        kw = {
            "num_splits": num_splits,
            "other_axis_partition": right_partitions,
            "manual_partition": bool(lengths),
            "_lengths": lengths,
        }
```

Reply:
In some cases, the control falls into `deploy_func_between_two_axis_partitions` that does not work correctly if `_lengths` or `manual_partition`  in kwargs.